htrc/torchlite/app.py

```diff
@@ -11,19 +14,29 @@
 from htrc.torchlite.filters import torchlite_stopword_filter, torchlite_stemmer, torchlite_lemmatizer
 from htrc.torchlite.middleware import TorchliteVersionHeaderMiddleware
 
-with open("config.yaml", mode="r", encoding="utf-8") as f:
-    config = safe_load(f)
+config_file: Optional[str] = os.getenv("TORCHLITE_CONFIG")
 
+if config_file is None:
+    raise OSError("Torchlite Configuration not found")
 
-def set_defaults(app: Torchlite, ef_api: Api) -> None:
-    if "featured_worksets_url" in config:
-        url = config["featured_worksets_url"]
-        if url:
-            r = requests.get(url)
-            if r.status_code == 200:
-                sample_workset_data = load(r.text, Loader=Loader)
-                for data in sample_workset_data:
-                    app.add_workset(**data)
+if config_file.startswith("http"):
+    resp: requests.Response = requests.get(config_file)
+    if resp.status_code == 200:
+        config = safe_load(resp.text)
+    else:
+        raise OSError(f"{config_file} not found")
+else:
+    p: Path = Path(config_file)
+    if p.exists():
```

Comment:
I'm not sure it's worth checking if p exists... just using open(...) directly will throw an exception indicating the issue just the same.

Reply:
In general I was just going for a quick solution here.  What about this: https://docs.pydantic.dev/latest/usage/settings/?

Reply:
I'm not familiar with pydantic... if it can help you load the config YAML easier, then sure, I have nothing against it.
I guess I don't know what that's used for...   but we should still use a `config.yaml` YAML-based configuration file.

Reply:
If you like type hinting you're going to love Pydantic: https://docs.pydantic.dev/latest/.  It underlies FastAPI, among other things.

Reply:
Since the code is working, I'm going to merge this pr but make a ticket: https://github.com/htrc/torchlite-backend/issues/39